covenant-signer/README.md

```diff
@@ -0,0 +1,335 @@
+  "iswatchonly": false,
+  "isscript": false,
+  "iswitness": true,
+  "witness_version": 0,
+  "witness_program": "e8bb0eba51e8daf4a5d27d44137dca0acb80face",
+  "pubkey": "023a79b546c79d7f7c5ff20620d914b5cf7250631d12f6e26427ed9d3f98c5ccb1",
+  "ischange": false,
+  "timestamp": 1732624709,
+  "hdkeypath": "m/84h/1h/0h/0/0",
+  "hdseedid": "0000000000000000000000000000000000000000",
+  "hdmasterfingerprint": "5e174bde",
+  "labels": [
+    ""
+  ]
+}
+```
+
+The most important field to focus on is `hdkeypath` that contains derivation path 
+of our key. In the example it is `84h/1h/0h/0/0` (the intilal `m/` can be ignored).
+
+Next, list all descriptors in the wallet, ensuring that private keys are included 
+in the output:
+
+```shell
+docker exec -it bitcoind bitcoin-cli -chain=regtest -rpcuser=user -rpcpassword=pass listdescriptors true
+```
+
+The terminal should produce output similar to the following:
+
+```json
+{
+  "wallet_name": "covenant-wallet",
+  "descriptors": [
+    {
+      "desc": "wpkh(tprv8ZgxMBicQKsPe9aCeUQgMEMy2YMZ6PHnn2iCuG12y5E8oYhYNEvUqUkNy6sJ7ViBmFUMicikHSK2LBUNPx5do5EDJBjG7puwd6azci2wEdq/84h/1h/0h/0/*)#sachkrde",
+      "timestamp": 1732624709,
+      "active": true,
+      "internal": false,
+      "range": [
+        0,
+        1000
+      ],
+      "next": 1,
+      "next_index": 1
+    }
+    ...
+  ]
+}
+
+```
+The most important field to note is the `desc` value:
+
+```json
+"desc": "wpkh(tprv8ZgxMBicQKsPe9aCeUQgMEMy2YMZ6PHnn2iCuG12y5E8oYhYNEvUqUkNy6sJ7ViBmFUMicikHSK2LBUNPx5do5EDJBjG7puwd6azci2wEdq/84h/1h/0h/0/*)#sachkrde"
+```
+
+Here, you can see the string starting with `tprv8ZgxMBicQKsPe9aCeUQgMEMy2YMZ6PHnn2iCuG12y5E8oYhYNEvUqUkNy6sJ7ViBmFUMicikHSK2LBUNPx5do5EDJBjG7puwd6azci2wEdq`
+is the **base58-encoded master private key** of the covenant wallet. 
+This key is critical for signing operations and should be securely stored.
+
+
+#### Deriving the Covenant Private Key from the Master Key
+
+You can derive the covenant private key from the master key by performing a 
+**BIP32 derivation**. The `covenant-signer`repository includes a command to 
+accomplish this:
+
+```shell
+covenant-signer derive-child-key \
+  tprv8ZgxMBicQKsPe9aCeUQgMEMy2YMZ6PHnn2iCuG12y5E8oYhYNEvUqUkNy6sJ7ViBmFUMicikHSK2LBUNPx5do5EDJBjG7puwd6azci2wEdq \
+  84h/1h/0h/0/0
+```
+
+The output will display the derived private and public keys:
+
+```shell
+Derived private key: fe1c56c494c730f13739c0655bf06e615409870200047fc65cdf781837cf7f06
+Derived public key: 023a79b546c79d7f7c5ff20620d914b5cf7250631d12f6e26427ed9d3f98c5ccb1
+```
+
+As seen, the **Derived Public Key**:
+
+```
+023a79b546c79d7f7c5ff20620d914b5cf7250631d12f6e26427ed9d3f98c5ccb1
+```
+
+Matches the public key obtained earlier using the `getaddressinfo` command.
+
+## 4. Importing the Derived Private Key into the Cosmos Keyring
```

Comment:
I would add here that:
- `file` backend should be used i.e private key will be stored in encrypted form on disk
-  when running `import-hex`  on encrypted file backend, caller will be asked for passphrase which will be later used to unlock the signer